### PR TITLE
Unify query part handling for rows and aggregates

### PR DESCRIPTION
**Pull request by soupi**
### What

fixes https://hasurahq.atlassian.net/browse/PG-112

Previously, order by and limit clauses were not applied for the aggregates part of the query as the spec said it was not needed, however, that is a mistake which we now fix.

### How

To fix this, we unify the way we handle rows and aggregates.
Callers will no longer need to call `translate_rows_query` and `translate_aggregates_query` separately and then try to stitch them together, but rather call one part, `translate_query`, which will call both.

Each translate part will call the same function, `translate_query_part` which translates the common elements (where, order by, limit, etc.), but will different in the way it handles fields or aggregates.

This way, we make sure that an ndc-spec query is translated consistently between the two parts - rows and aggregates.

### CODE REVIEW

File: crates/query-engine/translation/src/translation/query/mod.rs [hunk 1]
```diff
@@ -30,19 +30,15 @@ pub fn translate(
...
-        None,
-    )?;
 
-    let select_set = translate_query(
+    let select_set = root::translate_query(
         &env,
         &mut state,
-        &current_table,
-        &from_clause,
+        root::MakeFrom::Collection {
```
Comment (soupi) [inline crates/query-engine/translation/src/translation/query/mod.rs]: Instead of building a from clause and passing it, we build a structure describing how to build the from clause, so it can be built the same way for both `rows` and `aggregates` but produce different aliases using the state index.

File: crates/query-engine/translation/src/translation/query/relationships.rs [hunk 2]
```diff
@@ -35,87 +35,18 @@ pub fn translate_joins(
...
-            let select_set = super::translate_query(
+            let select_set = root::translate_query(
                 env,
                 state,
-                &target_collection,
-                &from_clause,
+                root::MakeFrom::Collection {
+                    name: relationship.target_collection.clone(),
+                    arguments: arguments.clone(),
+                },
```
Comment (soupi) [inline crates/query-engine/translation/src/translation/query/relationships.rs]: Instead of building a from clause and passing it, we build a structure describing how to build the from clause.

File: crates/query-engine/translation/src/translation/query/relationships.rs [hunk 3]
```diff
@@ -35,87 +35,22 @@ pub fn translate_joins(
...
+                // We ask to inject the join predicate into the where clause.
+                &Some(root::JoinPredicate {
+                    join_with: current_table,
+                    relationship,
+                }),
                 &join_field.query,
             )?;
 
-            // add join expressions to row / aggregate selects
```
Comment (soupi) [inline crates/query-engine/translation/src/translation/query/relationships.rs:56]: injecting the join predicate means we don't need to unpack afterwards.

File: crates/query-engine/translation/tests/goldenfiles/aggregate_limit_offset_order_by/request.json
Comment (soupi) [inline crates/query-engine/translation/tests/goldenfiles/aggregate_limit_offset_order_by/request.json:1]: Test that limit and order by has been applied to the aggregates query.

File: crates/query-engine/translation/tests/snapshots/tests__aggregate_limit_offset_order_by.snap [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                FROM
+                  "public"."Invoice" AS "%1_Invoice"
+                WHERE
+                  (
+                    "%1_Invoice"."BillingCountry" = cast($1 as "pg_catalog"."varchar")
+                  )
+                ORDER BY
+                  "%1_Invoice"."InvoiceId" ASC
+                LIMIT
+                  10 OFFSET 5
```
Comment (soupi) [inline crates/query-engine/translation/tests/snapshots/tests__aggregate_limit_offset_order_by.snap:42]: yep, has been applied.

File: crates/tests/databases-tests/src/aurora/snapshots/databases_tests__aurora__query_tests__aggregation__aggregate_count_albums_plus_field.snap
Comment (soupi) [inline crates/tests/databases-tests/src/aurora/snapshots/databases_tests__aurora__query_tests__aggregation__aggregate_count_albums_plus_field.snap:1]: Now we apply the limit. We also changed the test a bit so it is easy to verify.

File: crates/tests/databases-tests/src/aurora/snapshots/databases_tests__aurora__query_tests__sorting__select_order_by_artist_name.snap
Comment (soupi) [inline crates/tests/databases-tests/src/aurora/snapshots/databases_tests__aurora__query_tests__sorting__select_order_by_artist_name.snap:1]: the test request did not contain order for the album, so it was ambigious. fixed.

**Review by plcplc — APPROVED** [review]

File: changelog.md [hunk 5]
```diff
@@ -11,6 +11,9 @@
 
 ### Fixed
 
+- Unify query part handling for rows and aggregates.
```
[the remark this answers is not in the document]
Comment (soupi) [reply changelog.md:14]: Makes sense. Let me rephrase:

```suggestion
- Rows and aggregate parts of the query should operate on the same query parameters (where, order by, limit and offset).
```
